planner/core/planbuilder.go

```diff
@@ -723,20 +732,41 @@ func (b *PlanBuilder) buildAnalyzeTable(as *ast.AnalyzeTableStmt) (Plan, error)
 	return p, nil
 }
 
+// buildAuxColTasks builds auxiliary AnalyzeColumnsTask for column of single-column index,
+// because we need `NullCount` of the column histogram for single-column index histogram.
```

Comment:
If you only want the null count, can we build one more cop task to scan the null count of the index? When run `analyze index`, it does not need to scan the whole table.

Reply:
Updated, PTAL